polyfill/lib/calendar.mjs

```diff
@@ -348,6 +411,23 @@ impl['gregory'] = ObjectAssign({}, impl['iso8601'], {
       isoYear = gre.isoYear(fields.eraYear, fields.era);
     }
     return impl['iso8601'].yearMonthFromFields({ ...fields, year: isoYear }, overflow);
+  },
+  monthDayFromFields(fields, overflow) {
+    // Intentionally alphabetical
+    fields = ES.PrepareTemporalFields(fields, [
+      ['day'],
+      ['era', undefined],
+      ['eraYear', undefined],
+      ['month', undefined],
+      ['monthCode', undefined],
+      ['year', undefined]
+    ]);
+    if (fields.month !== undefined) gre.validateFields(fields);
```

Comment:
I believe we only want to `validateFields` (which actually only validates the correct combination of `year`/`era`/`eraYear` fields) if `month` is given (and `monthCode` is not, which I had forgotten and will amend). If we have `monthCode`, we don't care about whether there's a year, whereas with the other two types, we need to have a year unconditionally.

Reply:
Yes, but - we need to verify they don't conflict, even if we're only going to use one, as the calendar is responsible for validating the values of its fields.
It could also make sense to do that in validate fields; whichever you prefer.

Reply:
This `gregory` implementation chains up to the one in `iso8601`, so I believe it's OK. I've added a comment explaining what is going on.